pkg/skaffold/watch/triggers.go

```diff
@@ -125,3 +130,51 @@ func (t *manualTrigger) Start(ctx context.Context) (<-chan bool, error) {
 
 	return trigger, nil
 }
+
+// notifyTrigger watches for changes with fsnotify
+type fsNotifyTrigger struct {
+	Interval time.Duration
+}
+
+// Debounce tells the watcher to not debounce rapid sequence of changes.
+func (t *fsNotifyTrigger) Debounce() bool {
+	// This trigger has built-in debouncing.
+	return false
+}
+
+func (t *fsNotifyTrigger) WatchForChanges(out io.Writer) {
+	color.Yellow.Fprintln(out, "Watching for changes...")
+}
+
+// Start Listening for file system changes
+func (t *fsNotifyTrigger) Start(ctx context.Context) (<-chan bool, error) {
+	// TODO(@dgageot): If file changes happen too quickly, events might be lost
```

Comment:
we could compute the number of files in the watch directory at start up time and use that for the buffer size. just a thought